moon/controller_hauler_round2.py

```diff
@@ -88,45 +104,155 @@ def get_extra_status(self):
-                    pass
+            # FIND EXCAVATOR
             while True:
                 try:
-                    self.turn(normalizeAnglePIPI(self.turnto_angle - self.yaw), timeout=timedelta(seconds=15))
-                    self.wait(timedelta(seconds=3))
-                    break
-                except BusShutdownException:
-                    raise
-                except:
+                    self.look_for_rover()
+                except ChangeDriverException as e:
                     pass
 
-            self.finish_visually = True
+                # follow to ONHOLD distance
+                self.driving_mode = "onhold"
+                self.target_excavator_distance = EXCAVATOR_ONHOLD_GAP
+
+                # wait until rover is in position
+                while abs(self.target_excavator_distance - self.rover_distance) > DISTANCE_TOLERANCE:
+                    try:
+                        self.wait(timedelta(milliseconds=200))
+                    except ExcavatorLostException:
+                        continue
+
+                print(self.sim_time, self.robot_name, "Sending arrived message to excavator")
+                self.send_request('external_command excavator_1 arrived %.2f' % 0.0)
+
+                while self.set_yaw is None:
+                    try:
+                        self.wait(timedelta(seconds=1))
+                    except BusShutdownException:
+                        raise
+                    except ExcavatorLostException:
+                        continue
+                break
+
+
+            #self.set_brakes(True)
 
-            self.excavator_waiting = True
+            #self.send_request('vslam_reset', self.vslam_reset_time)
 
-            self.virtual_bumper = VirtualBumper(timedelta(seconds=10), 0.1) # need generous timeout because sideways moves are not sensed by bumper
+            #while not self.true_pose:
+            #    self.wait(timedelta(seconds=1))
+
+            #self.send_request('external_command excavator_1 pose %.1f %.1f' % (self.xyz[0], self.xyz[1]))
+
+            #while self.turnto_angle is None:
+            #    try:
+            #        self.wait(timedelta(seconds=1))
+            #    except BusShutdownException:
+            #        raise
+            #    except:
+            #        pass
+            #while True:
+            #    try:
+            #        self.turn(normalizeAnglePIPI(self.turnto_angle - self.yaw), timeout=timedelta(seconds=15))
+            #        self.wait(timedelta(seconds=3))
+            #        break
+            #    except BusShutdownException:
+            #        raise
+            #    except:
+            #        pass
+
+            #while self.excavator_yaw is None: # waiting for align command
+            #    try:
+            #        self.wait(timedelta(seconds=1))
+            #    except BusShutdownException:
+            #        raise
+            #    except:
+            #        pass
+
+
+            if False:
```

Comment:
is it intentionally "commented out code"?

Reply:
yes, I may still become handy later if we need to change the strategy. Eventually, once the current strategy proves itself longer term, we can do cleanup and remove all obsoleted code